x-pack/plugins/ingest_manager/server/services/epm/packages/install.ts

```diff
@@ -286,3 +302,29 @@ export const removeAssetsFromInstalledEsByType = async (
     installed_es: installedAssetsToSave,
   });
 };
+
+export async function ensurePackagesCompletedInstall(
+  savedObjectsClient: SavedObjectsClientContract,
+  callCluster: CallESAsCurrentUser
+) {
+  const installedSavedObjects = await getPackageSavedObjects(savedObjectsClient);
+  const installingPackages = installedSavedObjects.saved_objects.filter(
+    (pkg) => pkg.attributes.install_status === EpmPackageInstallStatus.installing
+  );
+  const installingPromises = installingPackages.reduce<Array<Promise<AssetReference[]>>>(
+    (acc, pkg) => {
+      const startDate = pkg.attributes.install_started_at;
+      const nowDate = new Date().toISOString();
+      const elapsedTime = Date.parse(nowDate) - Date.parse(startDate);
+      const pkgkey = `${pkg.attributes.name}-${pkg.attributes.install_version}`;
+      // reinstall package
+      if (elapsedTime > MAX_TIME_COMPLETE_INSTALL) {
```

Comment:
Following up on what I wrote above, I realized we can get this from ES by adding a `filter` to the `soClient.find` arguments. I think it would be more clear, but perhaps it's more expensive. Not a big deal either way, just a thought.